probe_scraper/runner.py

```diff
@@ -280,16 +312,26 @@ def load_glean_metrics(cache_dir, out_dir, repositories_file, dry_run, glean_rep
         for commit_hash, paths in commits.items():
             metrics_files = [p for p in paths if p.endswith(GLEAN_METRICS_FILENAME)]
             pings_files = [p for p in paths if p.endswith(GLEAN_PINGS_FILENAME)]
-
             try:
                 config = {"allow_reserved": repo_name.startswith("glean")}
+                for d in repositories:
+                    if vars(d)["name"] == repo_name:
+                        repo_dict = d.to_dict()
```

Comment:
I'm pretty sure this should work. Also, I think we should just use the name `repo` as opposed to `repo_dict`.

```suggestion
                repo = next(r for r in repositories if r.name == repo_name)
```